src/screens/sync/Main.tsx

```diff
@@ -0,0 +1,350 @@
+            <TouchableWithoutFeedback
+              onPress={
+                permissionEnabled ? undefined : () => Linking.openSettings()
+              }
+            >
+              <Text size="small" bold numberOfLines={1}>
+                {wifiConnected
+                  ? ssid || t(m.connectedNoNetworkName)
+                  : t(m.noConnection)}
+              </Text>
+            </TouchableWithoutFeedback>
+          </View>
+        </View>
+        <ViewSwitch value={viewMode} toggle={toggleViewMode} />
+      </View>
+      <View style={projectInfoCardStyles.bodyContainer}>
+        <View style={projectInfoCardStyles.logoImageContainer}>
+          <Image
+            // TODO: Use better res asset
+            source={require("../../../assets/cards.png")}
+            style={projectInfoCardStyles.logoImage}
+          />
+        </View>
+        <Text size="medium" bold>
+          {name}
+        </Text>
+      </View>
+    </View>
+  );
+};
+
+const animatedEllipsisStyles = StyleSheet.create({
+  container: {
+    flexDirection: "row",
+    alignItems: "baseline",
+  },
+  text: {
+    fontSize: textStyles.large.fontSize,
+    fontWeight: textStyles.bold.fontWeight,
+    // Hacky way of getting ellipsis to vertically align at bottom
+    marginBottom: -(textStyles.large.fontSize / 2),
+  },
+});
+
+const AnimatedEllipsis = () => {
+  const dots = React.useRef([
+    new Animated.Value(0),
+    new Animated.Value(0),
+    new Animated.Value(0),
+  ]).current;
+
+  React.useEffect(() => {
+    let mounted = true;
+
+    function animateDot(position: number) {
+      if (!mounted) return;
+
+      const targetDot = dots[position];
+      const nextDotPosition = (position + 1) % 3;
+
+      Animated.timing(targetDot, {
+        toValue: 1,
+        useNativeDriver: true,
+        duration: 350,
+      }).start(({ finished }) => {
+        if (finished && nextDotPosition === 0) {
+          dots.forEach((d) => {
+            d.setValue(0);
+          });
+        }
+        animateDot(nextDotPosition);
+      });
+    }
+
+    animateDot(0);
+
+    return () => {
+      mounted = false;
+      dots.forEach((d) => d.stopAnimation());
+    };
+  }, []);
+
+  return (
+    <View style={animatedEllipsisStyles.container}>
+      {dots.map((opacity, index) => (
+        <Animated.Text
+          key={index}
+          style={[animatedEllipsisStyles.text, { opacity }]}
+        >
+          .
+        </Animated.Text>
+      ))}
+    </View>
+  );
+};
+
+const devicesListStyles = StyleSheet.create({
+  container: {
+    paddingVertical: spacing.large,
+  },
+});
+
+const DevicesList = () => {
+  const { formatMessage: t } = useIntl();
+  const [status, setStatus] = React.useState<"loading" | "idle">("loading");
+
+  useFocusEffect(
+    React.useCallback(() => {
+      setStatus("loading");
+      setTimeout(() => {
+        setStatus("idle");
+      }, 3000);
+    }, [])
+  );
```

Comment:
using useFocusEffect here so that for the sake of testing, the loading state can be triggered by leaving the screen and re-entering